src/Locker.ts

```diff
@@ -31,16 +31,14 @@ export class Locker {
   private namespace: string
   private separator: string
 
-  constructor(lockerConfig: LockerConfig) {
-    const {driverNamespace, defaultDriverType, namespaceSeparator} = lockerConfig
-
-    this.setNamespace(driverNamespace, namespaceSeparator)
-    this.driver = defaultDriverType.isSupported() ? defaultDriverType : DRIVERS.MEMORY
+  constructor(private lockerConfig: LockerConfig) {
+    this.setNamespace(this.lockerConfig.driverNamespace, this.lockerConfig.namespaceSeparator)
+    this.driver = this.lockerConfig.defaultDriverType.isSupported() ? this.lockerConfig.defaultDriverType : DRIVERS.MEMORY
   }
 
-  public setNamespace(namespace, separator) {
+  public setNamespace(namespace: string, separator?: string) {
     this.namespace = namespace
-    this.separator = separator
+    this.separator = separator ? separator : this.lockerConfig.namespaceSeparator
```

Comment:
So something like (might be ternary as well)
```typescript
this.separator = separator && separator || this.separator && this.separator || this.lockerConfig.namespaceSeparator
```
Or it might be another method altogether and chain both methods
```typescript
Locker.setNamespace(namespace).setSeparator(separator)
```
It is a bit confusing to me, we are trying to set the namespace but at the same time we are setting the separator, how do i reset the separator to the default one (as a user)? Do i need to get an instance of the config and pass that?